beacon-chain/core/blocks/validity_conditions.go

```diff
@@ -0,0 +1,66 @@
+package blocks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/types"
+)
+
+// IsValidBlock ensures that the block is compliant with the block processing validity conditions.
+// Spec:
+//  For a beacon chain block, block, to be processed by a node, the following conditions must be met:
+//  The parent block with root block.parent_root has been processed and accepted.
+//  The node has processed its state up to slot, block.slot - 1.
+//  The Ethereum 1.0 block pointed to by the state.processed_pow_receipt_root has been processed and accepted.
+//  The node's local clock time is greater than or equal to state.genesis_time + block.slot * SLOT_DURATION.
+func IsValidBlock(
```

Comment:
Is this new or just moved from somewhere else?

Reply:
Moved to core/blocks/validity_conditions.go - it was previously in state